File: C#/TCPServer.cs
//Includes
 using System;
 using System.Text;
 using System.Net;
 using System.Net.Sockets;
public class TCPServer{
 public static void Main(){
  byte[] buffer=new byte[256];
  Socket server=new Socket(AddressFamily.InterNetwork,SocketType.Stream,ProtocolType.Tcp);
  IPEndPoint address=new IPEndPoint(IPAddress.Any,3721);
  server.Bind(address);
  server.Listen(5);
  while(true){
   Socket client=server.Accept();
   int size=client.Receive(buffer,0,255,0,out SocketError error);
   buffer[size]=0;
   Console.Write(Encoding.ASCII.GetString(buffer));
   client.Send(Encoding.ASCII.GetBytes("Message received\n"));
   client.Close();}}}

File: Zig/TCPServer.zig
const std = @import("std");
pub fn main()!void{
 var buffer:[256]u8=undefined;
 var server=std.net.StreamServer.init(.{.reuse_address=true });
 try server.listen(std.net.Address.initIp4([4]u8{0,0,0,0},3721));
 while (true){
  var client=try server.accept();
  var size=try client.stream.read(&buffer);
  std.debug.print("{s}",.{buffer[0..size]});
  size=try client.stream.write("Message received\n");
  client.stream.close();}}